diff -upr mdm-1.1.1-orig/data/greeter-autostart/Makefile.am mdm-1.1.1/data/greeter-autostart/Makefile.am
--- mdm-1.1.1-orig/data/greeter-autostart/Makefile.am	2012-02-04 13:19:11.000000000 +0100
+++ mdm-1.1.1/data/greeter-autostart/Makefile.am	2012-04-19 13:13:10.000000000 +0200
@@ -18,7 +18,6 @@ desktop_in_files = 	\
 	mate-mag.desktop.in			\
 	mate-power-manager.desktop.in		\
 	mate-settings-daemon.desktop.in	\
-	gok.desktop.in				\
 	marco.desktop.in			\
 	orca-screen-reader.desktop.in		\
 	polkit-mate-authentication-agent-1.desktop.in	\
