diff -upr mdm-1.1.1-orig/data/greeter-autostart/Makefile.am mdm-1.1.1/data/greeter-autostart/Makefile.am
--- mdm-1.1.1-orig/data/greeter-autostart/Makefile.am	2012-02-04 13:19:11.000000000 +0100
+++ mdm-1.1.1/data/greeter-autostart/Makefile.am	2012-04-19 13:53:24.000000000 +0200
@@ -15,7 +15,6 @@ desktopdir = $(pkgdatadir)/autostart/Log
 desktop_in_files = 	\
 	at-spi-registryd-wrapper.desktop.in	\
 	mdm-simple-greeter.desktop.in		\
-	mate-mag.desktop.in			\
 	mate-power-manager.desktop.in		\
 	mate-settings-daemon.desktop.in	\
 	gok.desktop.in				\
