diff -upr mate-themes-1.2.2-orig/desktop-themes/Aldabra/gtk-2.0/assets/Makefile.am mate-themes-1.2.2/desktop-themes/Aldabra/gtk-2.0/assets/Makefile.am
--- mate-themes-1.2.2-orig/desktop-themes/Aldabra/gtk-2.0/assets/Makefile.am	2012-04-04 17:57:06.000000000 +0200
+++ mate-themes-1.2.2/desktop-themes/Aldabra/gtk-2.0/assets/Makefile.am	2012-04-28 12:13:08.000000000 +0200
@@ -1,5 +1,5 @@
 
-themedir = $(datadir)/themes/Aldabra/gtk-2.0/assets
+themedir = $(datadir)/themes/Aldabras/gtk-2.0/assets
 
 theme_DATA = \
 	checkbox-checked-insensitive.svg  \
diff -upr mate-themes-1.2.2-orig/desktop-themes/Aldabra/gtk-2.0/Makefile.am mate-themes-1.2.2/desktop-themes/Aldabra/gtk-2.0/Makefile.am
--- mate-themes-1.2.2-orig/desktop-themes/Aldabra/gtk-2.0/Makefile.am	2012-04-04 17:57:06.000000000 +0200
+++ mate-themes-1.2.2/desktop-themes/Aldabra/gtk-2.0/Makefile.am	2012-04-28 12:08:48.000000000 +0200
@@ -1,5 +1,5 @@
 
-themedir = $(datadir)/themes/Aldabra/gtk-2.0
+themedir = $(datadir)/themes/Aldabras/gtk-2.0
 
 theme_DATA = gtkrc
 
diff -upr mate-themes-1.2.2-orig/desktop-themes/Aldabra/gtk-3.0/assets/Makefile.am mate-themes-1.2.2/desktop-themes/Aldabra/gtk-3.0/assets/Makefile.am
--- mate-themes-1.2.2-orig/desktop-themes/Aldabra/gtk-3.0/assets/Makefile.am	2012-04-04 17:57:06.000000000 +0200
+++ mate-themes-1.2.2/desktop-themes/Aldabra/gtk-3.0/assets/Makefile.am	2012-04-28 12:13:26.000000000 +0200
@@ -1,5 +1,5 @@
 
-themedir = $(datadir)/themes/Aldabra/gtk-3.0/assets
+themedir = $(datadir)/themes/Aldabras/gtk-3.0/assets
 
 theme_DATA = \
 	checkbox-checked-insensitive.svg \
diff -upr mate-themes-1.2.2-orig/desktop-themes/Aldabra/gtk-3.0/Makefile.am mate-themes-1.2.2/desktop-themes/Aldabra/gtk-3.0/Makefile.am
--- mate-themes-1.2.2-orig/desktop-themes/Aldabra/gtk-3.0/Makefile.am	2012-04-04 17:57:06.000000000 +0200
+++ mate-themes-1.2.2/desktop-themes/Aldabra/gtk-3.0/Makefile.am	2012-04-28 12:09:06.000000000 +0200
@@ -1,5 +1,5 @@
 
-themedir = $(datadir)/themes/Aldabra/gtk-3.0
+themedir = $(datadir)/themes/Aldabras/gtk-3.0
 
 theme_DATA = \
 	settings.ini \
diff -upr mate-themes-1.2.2-orig/desktop-themes/Aldabra/Makefile.am mate-themes-1.2.2/desktop-themes/Aldabra/Makefile.am
--- mate-themes-1.2.2-orig/desktop-themes/Aldabra/Makefile.am	2012-04-04 17:57:06.000000000 +0200
+++ mate-themes-1.2.2/desktop-themes/Aldabra/Makefile.am	2012-04-28 12:08:08.000000000 +0200
@@ -1,4 +1,4 @@
-THEME_NAME=Aldabra
+THEME_NAME=Aldabras
  
 themedir = $(datadir)/themes/$(THEME_NAME)
 
@@ -20,5 +20,5 @@ CLEANFILES = index.theme
 
 index.theme: index.theme.in
 	$(AM_V_GEN)sed \
-		-e s!\@BACKGROUND_DIR\@!$(datadir)/pixmaps/backgrounds/mate/abstract! \
+		-e s!\@BACKGROUND_DIR\@!$(datadir)/backgrounds/mate/abstract! \
 		< $< > $@
diff -upr mate-themes-1.2.2-orig/desktop-themes/Aldabra/metacity-1/Makefile.am mate-themes-1.2.2/desktop-themes/Aldabra/metacity-1/Makefile.am
--- mate-themes-1.2.2-orig/desktop-themes/Aldabra/metacity-1/Makefile.am	2012-04-04 17:57:06.000000000 +0200
+++ mate-themes-1.2.2/desktop-themes/Aldabra/metacity-1/Makefile.am	2012-04-28 12:10:00.000000000 +0200
@@ -1,5 +1,5 @@
 
-themedir = $(datadir)/themes/Aldabra/metacity-1
+themedir = $(datadir)/themes/Aldabras/metacity-1
 
 theme_DATA = \
 	metacity-theme-1.xml \
diff -upr mate-themes-1.2.2-orig/desktop-themes/Aldabra/metacity-1/metacity-theme-1.xml mate-themes-1.2.2/desktop-themes/Aldabra/metacity-1/metacity-theme-1.xml
--- mate-themes-1.2.2-orig/desktop-themes/Aldabra/metacity-1/metacity-theme-1.xml	2012-04-04 17:57:06.000000000 +0200
+++ mate-themes-1.2.2/desktop-themes/Aldabra/metacity-1/metacity-theme-1.xml	2012-04-28 11:52:00.000000000 +0200
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 <metacity_theme>
 	<info>
-		<name>Aldabra</name>
+		<name>Aldabras</name>
 		<author>Hylke Bons</author>
 		<copyright>Intel, 2010</copyright>
 		<date>April 23, 2010</date>
diff -upr mate-themes-1.2.2-orig/desktop-themes/Aldabra/metacity-1/metacity-theme-2.xml mate-themes-1.2.2/desktop-themes/Aldabra/metacity-1/metacity-theme-2.xml
--- mate-themes-1.2.2-orig/desktop-themes/Aldabra/metacity-1/metacity-theme-2.xml	2012-04-04 17:57:06.000000000 +0200
+++ mate-themes-1.2.2/desktop-themes/Aldabra/metacity-1/metacity-theme-2.xml	2012-04-28 11:51:58.000000000 +0200
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 <metacity_theme>
 <info>
-	<name>Aldabra</name>
+	<name>Aldabras</name>
 	<author>GNOME Art Team &lt;art.gnome.org&gt;</author>
 	<copyright>&#194; Intel, &#194; Red Hat, Lapo Calamandrei</copyright>
 	<date>2010</date>
diff -upr mate-themes-1.2.2-orig/desktop-themes/Aldabra/metacity-1/metacity-theme-3.xml mate-themes-1.2.2/desktop-themes/Aldabra/metacity-1/metacity-theme-3.xml
--- mate-themes-1.2.2-orig/desktop-themes/Aldabra/metacity-1/metacity-theme-3.xml	2012-04-04 17:57:06.000000000 +0200
+++ mate-themes-1.2.2/desktop-themes/Aldabra/metacity-1/metacity-theme-3.xml	2012-04-28 11:51:56.000000000 +0200
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 <metacity_theme>
 <info>
-	<name>Aldabra</name>
+	<name>Aldabras</name>
 	<author>GNOME Art Team &lt;art.gnome.org&gt;</author>
 	<copyright>&#194; Intel, &#194; Red Hat, Lapo Calamandrei</copyright>
 	<date>2010</date>
