diff -upr mate-themes-1.2.2-orig/desktop-themes/Aldabra/index.theme.in mate-themes-1.2.2/desktop-themes/Aldabra/index.theme.in
--- mate-themes-1.2.2-orig/desktop-themes/Aldabra/index.theme.in	2012-04-04 17:57:06.000000000 +0200
+++ mate-themes-1.2.2/desktop-themes/Aldabra/index.theme.in	2012-05-03 23:44:39.939084074 +0200
@@ -1,34 +1,13 @@
 [X-GNOME-Metatheme]
-Name=Aldabra
+Name=Aldabras
 Type=X-GNOME-Metatheme
 Encoding=UTF-8
-GtkTheme=Aldabra
-MetacityTheme=Aldabra
-CursorTheme=mate
-CursorSize=24
-NotificationTheme=slider
-ApplicationFont=Cantarell 11
-DocumentFont=Sans 11
-DesktopFont=Cantarell 11
-WindowTitleFont=Cantarell Bold 11
-FixedWidthFont=Monospace 9
+GtkTheme=Aldabras
+MetacityTheme=Aldabras
+IconTheme=mate
 
 [Desktop Entry]
-Name=Aldabra
-Type=X-MATE-Metatheme
-Comment=
+Name=Aldabras
+Type=X-GNOME-Metatheme
+Comment=gtk2/3 adwaita theme
 
-[X-MATE-Metatheme]
-Name=Aldabra
-GtkTheme=Aldabra
-MarcoTheme=Aldabra
-IconTheme=mate
-CursorTheme=mate
-CursorSize=24
-NotificationTheme=slider
-ApplicationFont=Cantarell 11
-DocumentFont=Sans 11
-DesktopFont=Cantarell 11
-WindowTitleFont=Cantarell Bold 11
-FixedWidthFont=Monospace 9
-BackgroundImage=@BACKGROUND_DIR@/stripes.jpg
