diff -upr mate-menu-editor-2011.12.01-orig/Alacarte/MainWindow.py mate-menu-editor-2011.12.01/Alacarte/MainWindow.py
--- mate-menu-editor-2011.12.01-orig/Alacarte/MainWindow.py	2011-12-14 11:04:30.000000000 +0100
+++ mate-menu-editor-2011.12.01/Alacarte/MainWindow.py	2012-03-03 12:23:25.000000000 +0100
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-#   Alacarte Menu Editor - Simple fd.o Compliant Menu Editor
+#   MateAlacarte Menu Editor - Simple fd.o Compliant Menu Editor
 #   Copyright (C) 2006  Travis Watkins
 #
 #   This library is free software; you can redistribute it and/or
@@ -22,14 +22,14 @@ import gettext
 import subprocess
 import urllib
 try:
-	from Alacarte import config
+	from MateAlacarte import config
 	gettext.bindtextdomain(config.GETTEXT_PACKAGE,config.localedir)
 	gettext.textdomain(config.GETTEXT_PACKAGE)
 except:
 	pass
 _ = gettext.gettext
-from Alacarte.MenuEditor import MenuEditor
-from Alacarte import util
+from MateAlacarte.MenuEditor import MenuEditor
+from MateAlacarte import util
 
 class MainWindow:
 	timer = None
@@ -46,10 +46,10 @@ class MainWindow:
 		self.file_path = datadir
 		self.version = version
 		self.editor = MenuEditor()
-		gtk.window_set_default_icon_name('alacarte')
+		gtk.window_set_default_icon_name('matealacarte')
 		self.tree = gtk.Builder()
 		self.tree.set_translation_domain(config.GETTEXT_PACKAGE)
-		self.tree.add_from_file(os.path.join(self.file_path, 'alacarte.ui'))
+		self.tree.add_from_file(os.path.join(self.file_path, 'matealacarte.ui'))
 		self.tree.connect_signals(self)
 		self.setupMenuTree()
 		self.setupItemTree()
@@ -294,7 +294,7 @@ class MainWindow:
 			menu_tree.get_selection().select_path((0,))
 		else:
 			parent = menus[iter][2]
-		file_path = os.path.join(util.getUserDirectoryPath(), util.getUniqueFileId('alacarte-made', '.directory'))
+		file_path = os.path.join(util.getUserDirectoryPath(), util.getUniqueFileId('matealacarte-made', '.directory'))
 		process = subprocess.Popen(['mate-desktop-item-edit', file_path], env=os.environ)
 		gobject.timeout_add(100, self.waitForNewMenuProcess, process, parent.menu_id, file_path)
 
@@ -307,7 +307,7 @@ class MainWindow:
 			menu_tree.get_selection().select_path((0,))
 		else:
 			parent = menus[iter][2]
-		file_path = os.path.join(util.getUserItemPath(), util.getUniqueFileId('alacarte-made', '.desktop'))
+		file_path = os.path.join(util.getUserItemPath(), util.getUniqueFileId('matealacarte-made', '.desktop'))
 		process = subprocess.Popen(['mate-desktop-item-edit', file_path], env=os.environ)
 		gobject.timeout_add(100, self.waitForNewItemProcess, process, parent.menu_id, file_path)
 
@@ -361,7 +361,7 @@ class MainWindow:
 			file_type = 'Item'
 		elif item.get_type() == matemenu.TYPE_DIRECTORY:
 			if item.get_desktop_file_path() == None:
-				file_path = util.getUniqueFileId('alacarte-made', '.directory')
+				file_path = util.getUniqueFileId('matealacarte-made', '.directory')
 				parser = util.DesktopParser(file_path, 'Directory')
 				parser.set('Name', item.get_name())
 				parser.set('Comment', item.get_comment())
@@ -545,8 +545,8 @@ class MainWindow:
 			content_type = file_info.get_content_type()
 			if content_type == 'application/x-desktop':
 				input_stream = myfile.read()
-				open('/tmp/alacarte-dnd.desktop', 'w').write(input_stream.read())
-				parser = util.DesktopParser('/tmp/alacarte-dnd.desktop')
+				open('/tmp/matealacarte-dnd.desktop', 'w').write(input_stream.read())
+				parser = util.DesktopParser('/tmp/matealacarte-dnd.desktop')
 				self.editor.createItem(parent, parser.get('Icon'), parser.get('Name', self.editor.locale), parser.get('Comment', self.editor.locale), parser.get('Exec'), parser.get('Terminal'), before, after)
 			elif content_type in ('application/x-shellscript', 'application/x-executable'):
 				self.editor.createItem(parent, None, os.path.split(file_path)[1].strip(), None, file_path.replace('file://', '').strip(), False, before, after)
diff -upr mate-menu-editor-2011.12.01-orig/Alacarte/Makefile.am mate-menu-editor-2011.12.01/Alacarte/Makefile.am
--- mate-menu-editor-2011.12.01-orig/Alacarte/Makefile.am	2011-12-14 11:04:30.000000000 +0100
+++ mate-menu-editor-2011.12.01/Alacarte/Makefile.am	2012-03-03 12:23:38.000000000 +0100
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-appdir = $(pythondir)/Alacarte
+appdir = $(pythondir)/MateAlacarte
 app_PYTHON = __init__.py MainWindow.py MenuEditor.py util.py
 nodist_app_PYTHON = config.py
 
diff -upr mate-menu-editor-2011.12.01-orig/Alacarte/Makefile.in mate-menu-editor-2011.12.01/Alacarte/Makefile.in
--- mate-menu-editor-2011.12.01-orig/Alacarte/Makefile.in	2011-12-14 11:04:30.000000000 +0100
+++ mate-menu-editor-2011.12.01/Alacarte/Makefile.in	2012-03-03 12:23:36.000000000 +0100
@@ -197,7 +197,7 @@ target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-appdir = $(pythondir)/Alacarte
+appdir = $(pythondir)/MateAlacarte
 app_PYTHON = __init__.py MainWindow.py MenuEditor.py util.py
 nodist_app_PYTHON = config.py
 CLEANFILES = config.py
diff -upr mate-menu-editor-2011.12.01-orig/Alacarte/MenuEditor.py mate-menu-editor-2011.12.01/Alacarte/MenuEditor.py
--- mate-menu-editor-2011.12.01-orig/Alacarte/MenuEditor.py	2011-12-14 11:04:30.000000000 +0100
+++ mate-menu-editor-2011.12.01/Alacarte/MenuEditor.py	2012-03-03 12:23:33.000000000 +0100
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-#   Alacarte Menu Editor - Simple fd.o Compliant Menu Editor
+#   MateAlacarte Menu Editor - Simple fd.o Compliant Menu Editor
 #   Copyright (C) 2006  Travis Watkins, Heinrich Wendel
 #
 #   This library is free software; you can redistribute it and/or
@@ -18,7 +18,7 @@
 
 import os, re, xml.dom.minidom, locale
 import matemenu
-from Alacarte import util
+from MateAlacarte import util
 
 class Menu:
 	tree = None
diff -upr mate-menu-editor-2011.12.01-orig/Alacarte/util.py mate-menu-editor-2011.12.01/Alacarte/util.py
--- mate-menu-editor-2011.12.01-orig/Alacarte/util.py	2011-12-14 11:04:30.000000000 +0100
+++ mate-menu-editor-2011.12.01/Alacarte/util.py	2012-03-03 12:23:30.000000000 +0100
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-#   Alacarte Menu Editor - Simple fd.o Compliant Menu Editor
+#   MateAlacarte Menu Editor - Simple fd.o Compliant Menu Editor
 #   Copyright (C) 2006  Travis Watkins
 #
 #   This library is free software; you can redistribute it and/or
diff -upr mate-menu-editor-2011.12.01-orig/alacarte.in mate-menu-editor-2011.12.01/alacarte.in
--- mate-menu-editor-2011.12.01-orig/alacarte.in	2011-12-14 11:04:30.000000000 +0100
+++ mate-menu-editor-2011.12.01/alacarte.in	2012-03-03 12:32:34.000000000 +0100
@@ -1,7 +1,7 @@
 #! @PYTHON@ @PYOPTIONS@
 # -*- python -*-
 # -*- coding: utf-8 -*-
-#   Alacarte Menu Editor - Simple fd.o Compliant Menu Editor
+#   MateAlacarte Menu Editor - Simple fd.o Compliant Menu Editor
 #   Copyright (C) 2006  Travis Watkins
 #
 #   This library is free software; you can redistribute it and/or
@@ -19,11 +19,11 @@
 #   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 
 import sys
-from Alacarte.MainWindow import MainWindow
+from MateAlacarte.MainWindow import MainWindow
 
 def main():
 	try:
-		from Alacarte import config
+		from MateAlacarte import config
 		datadir = config.pkgdatadir
 		version = config.VERSION
 	except:
diff -upr mate-menu-editor-2011.12.01-orig/configure mate-menu-editor-2011.12.01/configure
--- mate-menu-editor-2011.12.01-orig/configure	2012-01-08 07:03:54.000000000 +0100
+++ mate-menu-editor-2011.12.01/configure	2012-03-03 12:32:35.000000000 +0100
@@ -1,8 +1,8 @@
 #! /bin/sh
 # Guess values for system-dependent variables and create Makefiles.
-# Generated by GNU Autoconf 2.68 for alacarte 0.13.2.
+# Generated by GNU Autoconf 2.68 for matealacarte 0.13.2.
 #
-# Report bugs to <http://bugzilla.mate.org/enter_bug.cgi?product=alacarte>.
+# Report bugs to <http://bugzilla.mate.org/enter_bug.cgi?product=matealacarte>.
 #
 #
 # Copyright (C) 1992, 1993, 1994, 1995, 1996, 1998, 1999, 2000, 2001,
@@ -239,7 +239,7 @@ fi
     $as_echo "$0: be upgraded to zsh 4.3.4 or later."
   else
     $as_echo "$0: Please tell bug-autoconf@gnu.org and
-$0: http://bugzilla.mate.org/enter_bug.cgi?product=alacarte
+$0: http://bugzilla.mate.org/enter_bug.cgi?product=matealacarte
 $0: about your system, including any error possibly output
 $0: before this message. Then install a modern shell, or
 $0: manually run the script under such a shell if you do
@@ -559,11 +559,11 @@ MFLAGS=
 MAKEFLAGS=
 
 # Identity of this package.
-PACKAGE_NAME='alacarte'
-PACKAGE_TARNAME='alacarte'
+PACKAGE_NAME='matealacarte'
+PACKAGE_TARNAME='matealacarte'
 PACKAGE_VERSION='0.13.2'
-PACKAGE_STRING='alacarte 0.13.2'
-PACKAGE_BUGREPORT='http://bugzilla.mate.org/enter_bug.cgi?product=alacarte'
+PACKAGE_STRING='matealacarte 0.13.2'
+PACKAGE_BUGREPORT='http://bugzilla.mate.org/enter_bug.cgi?product=matealacarte'
 PACKAGE_URL=''
 
 ac_unique_file="alacarte.in"
@@ -1313,7 +1313,7 @@ if test "$ac_init_help" = "long"; then
   # Omit some internal or obsolete options to make the list less imposing.
   # This message is too long to be a string in the A/UX 3.1 sh.
   cat <<_ACEOF
-\`configure' configures alacarte 0.13.2 to adapt to many kinds of systems.
+\`configure' configures matealacarte 0.13.2 to adapt to many kinds of systems.
 
 Usage: $0 [OPTION]... [VAR=VALUE]...
 
@@ -1361,7 +1361,7 @@ Fine tuning of the installation director
   --infodir=DIR           info documentation [DATAROOTDIR/info]
   --localedir=DIR         locale-dependent data [DATAROOTDIR/locale]
   --mandir=DIR            man documentation [DATAROOTDIR/man]
-  --docdir=DIR            documentation root [DATAROOTDIR/doc/alacarte]
+  --docdir=DIR            documentation root [DATAROOTDIR/doc/matealacarte]
   --htmldir=DIR           html documentation [DOCDIR]
   --dvidir=DIR            dvi documentation [DOCDIR]
   --pdfdir=DIR            pdf documentation [DOCDIR]
@@ -1379,7 +1379,7 @@ fi
 
 if test -n "$ac_init_help"; then
   case $ac_init_help in
-     short | recursive ) echo "Configuration of alacarte 0.13.2:";;
+     short | recursive ) echo "Configuration of matealacarte 0.13.2:";;
    esac
   cat <<\_ACEOF
 
@@ -1415,7 +1415,7 @@ Some influential environment variables:
 Use these variables to override the choices made by `configure' or to help
 it to find libraries and programs with nonstandard names/locations.
 
-Report bugs to <http://bugzilla.mate.org/enter_bug.cgi?product=alacarte>.
+Report bugs to <http://bugzilla.mate.org/enter_bug.cgi?product=matealacarte>.
 _ACEOF
 ac_status=$?
 fi
@@ -1478,7 +1478,7 @@ fi
 test -n "$ac_init_help" && exit $ac_status
 if $ac_init_version; then
   cat <<\_ACEOF
-alacarte configure 0.13.2
+matealacarte configure 0.13.2
 generated by GNU Autoconf 2.68
 
 Copyright (C) 2010 Free Software Foundation, Inc.
@@ -1680,7 +1680,7 @@ $as_echo "$as_me: WARNING: $2:     secti
     { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: $2: proceeding with the compiler's result" >&5
 $as_echo "$as_me: WARNING: $2: proceeding with the compiler's result" >&2;}
 ( $as_echo "## ---------------------------------------------------------------------- ##
-## Report this to http://bugzilla.mate.org/enter_bug.cgi?product=alacarte ##
+## Report this to http://bugzilla.mate.org/enter_bug.cgi?product=matealacarte ##
 ## ---------------------------------------------------------------------- ##"
      ) | sed "s/^/$as_me: WARNING:     /" >&2
     ;;
@@ -1847,7 +1847,7 @@ cat >config.log <<_ACEOF
 This file contains any messages produced by compilers while
 running configure, to aid debugging if configure makes a mistake.
 
-It was created by alacarte $as_me 0.13.2, which was
+It was created by matealacarte $as_me 0.13.2, which was
 generated by GNU Autoconf 2.68.  Invocation command line was
 
   $ $0 $@
@@ -2664,7 +2664,7 @@ fi
 
 
 # Define the identity of the package.
- PACKAGE='alacarte'
+ PACKAGE='matealacarte'
  VERSION='0.13.2'
 
 
@@ -2735,7 +2735,7 @@ fi
 ACLOCAL="$ACLOCAL $ACLOCAL_FLAGS"
 
 
-GETTEXT_PACKAGE=alacarte
+GETTEXT_PACKAGE=matealacarte
 
 
 cat >>confdefs.h <<_ACEOF
@@ -6819,7 +6819,7 @@ cat >>$CONFIG_STATUS <<\_ACEOF || ac_wri
 # report actual input values of CONFIG_FILES etc. instead of their
 # values after options handling.
 ac_log="
-This file was extended by alacarte $as_me 0.13.2, which was
+This file was extended by matealacarte $as_me 0.13.2, which was
 generated by GNU Autoconf 2.68.  Invocation command line was
 
   CONFIG_FILES    = $CONFIG_FILES
@@ -6879,13 +6879,13 @@ $config_headers
 Configuration commands:
 $config_commands
 
-Report bugs to <http://bugzilla.mate.org/enter_bug.cgi?product=alacarte>."
+Report bugs to <http://bugzilla.mate.org/enter_bug.cgi?product=matealacarte>."
 
 _ACEOF
 cat >>$CONFIG_STATUS <<_ACEOF || ac_write_fail=1
 ac_cs_config="`$as_echo "$ac_configure_args" | sed 's/^ //; s/[\\""\`\$]/\\\\&/g'`"
 ac_cs_version="\\
-alacarte config.status 0.13.2
+matealacarte config.status 0.13.2
 configured by $0, generated by GNU Autoconf 2.68,
   with options \\"\$ac_cs_config\\"
 
diff -upr mate-menu-editor-2011.12.01-orig/configure.ac mate-menu-editor-2011.12.01/configure.ac
--- mate-menu-editor-2011.12.01-orig/configure.ac	2012-01-08 06:59:02.000000000 +0100
+++ mate-menu-editor-2011.12.01/configure.ac	2012-03-03 12:32:34.000000000 +0100
@@ -1,7 +1,7 @@
 dnl -*- Mode: autoconf -*-
 dnl Process this file with autoconf to produce a configure script.
 AC_PREREQ(2.53)
-AC_INIT(alacarte, 2011.12.01, http://bugzilla.mate.org/enter_bug.cgi?product=alacarte)
+AC_INIT(matealacarte, 2011.12.01, http://bugzilla.mate.org/enter_bug.cgi?product=matealacarte)
 AC_CONFIG_SRCDIR(alacarte.in)
 AC_CONFIG_HEADERS([config.h])
 AM_INIT_AUTOMAKE([1.7])
@@ -14,7 +14,7 @@ dnl put the ACLOCAL flags in the makefil
 ACLOCAL="$ACLOCAL $ACLOCAL_FLAGS"
 
 
-GETTEXT_PACKAGE=alacarte
+GETTEXT_PACKAGE=matealacarte
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [Gettext package])
 AM_GLIB_GNU_GETTEXT
diff -upr mate-menu-editor-2011.12.01-orig/data/alacarte.desktop.in.in mate-menu-editor-2011.12.01/data/alacarte.desktop.in.in
--- mate-menu-editor-2011.12.01-orig/data/alacarte.desktop.in.in	2011-12-14 11:04:30.000000000 +0100
+++ mate-menu-editor-2011.12.01/data/alacarte.desktop.in.in	2012-03-03 13:11:33.000000000 +0100
@@ -1,7 +1,7 @@
 [Desktop Entry]
 _Name=Main Menu
 _Comment=Change which applications are shown on the main menu
-Exec=alacarte
+Exec=matealacarte
 Terminal=false
 Type=Application
 StartupNotify=false
@@ -9,6 +9,6 @@ Categories=MATE;Settings;DesktopSettings
 NotShowIn=KDE;
 Icon=alacarte
 X-MATE-Bugzilla-Bugzilla=MATE
-X-MATE-Bugzilla-Product=alacarte
+X-MATE-Bugzilla-Product=matealacarte
 X-MATE-Bugzilla-Component=general
 X-MATE-Bugzilla-Version=@VERSION@
diff -upr mate-menu-editor-2011.12.01-orig/data/icons/Makefile.am mate-menu-editor-2011.12.01/data/icons/Makefile.am
--- mate-menu-editor-2011.12.01-orig/data/icons/Makefile.am	2011-12-14 11:04:30.000000000 +0100
+++ mate-menu-editor-2011.12.01/data/icons/Makefile.am	2012-03-03 12:27:43.000000000 +0100
@@ -33,12 +33,12 @@ update-icon-cache:
 	@-if test -z "$(DESTDIR)"; then \
 		echo "Updating Gtk icon cache."; \
 		for theme in $(public_icons_themes); do \
-			$(gtk_update_icon_cache) $(datadir)/icons/$$theme; \
+			$(gtk_update_icon_cache) $(datadir)/icons/mate; \
 		done; \
 	else \
 		echo "*** Icon cache not updated.  After (un)install, run this:"; \
 		for theme in $(public_icons_themes); do \
-			echo "***   $(gtk_update_icon_cache) $(datadir)/icons/$$theme"; \
+			echo "***   $(gtk_update_icon_cache) $(datadir)/icons/mate"; \
 		done; \
 	fi
 
@@ -48,16 +48,16 @@ install-icons:
 		CONTEXT=`echo $$icon | cut -d_ -f2`; \
 		SIZE=`echo $$icon | cut -d_ -f3`; \
 		ICONFILE=`echo $$icon | cut -d_ -f4`; \
-		mkdir -p $(DESTDIR)$(datadir)/icons/$$THEME/$$SIZE/$$CONTEXT; \
-		$(INSTALL_DATA) $(srcdir)/$$icon $(DESTDIR)$(datadir)/icons/$$THEME/$$SIZE/$$CONTEXT/$$ICONFILE; \
+		mkdir -p $(DESTDIR)$(datadir)/icons/mate/$$SIZE/$$CONTEXT; \
+		$(INSTALL_DATA) $(srcdir)/$$icon $(DESTDIR)$(datadir)/icons/mate/$$SIZE/$$CONTEXT/$$ICONFILE; \
 	done; \
 	for icon in $(private_icons); do \
 		THEME=`echo $$icon | cut -d_ -f1`; \
 		CONTEXT=`echo $$icon | cut -d_ -f2`; \
 		SIZE=`echo $$icon | cut -d_ -f3`; \
 		ICONFILE=`echo $$icon | cut -d_ -f4`; \
-		mkdir -p $(DESTDIR)$(pkgdatadir)/icons/$$THEME/$$SIZE/$$CONTEXT; \
-		$(INSTALL_DATA) $(srcdir)/$$icon $(DESTDIR)$(pkgdatadir)/icons/$$THEME/$$SIZE/$$CONTEXT/$$ICONFILE; \
+		mkdir -p $(DESTDIR)$(pkgdatadir)/icons/mate/$$SIZE/$$CONTEXT; \
+		$(INSTALL_DATA) $(srcdir)/$$icon $(DESTDIR)$(pkgdatadir)/icons/mate/$$SIZE/$$CONTEXT/$$ICONFILE; \
 	done
 
 uninstall-icons:
@@ -66,14 +66,14 @@ uninstall-icons:
 		CONTEXT=`echo $$icon | cut -d_ -f2`; \
 		SIZE=`echo $$icon | cut -d_ -f3`; \
 		ICONFILE=`echo $$icon | cut -d_ -f4`; \
-		rm -f $(DESTDIR)$(datadir)/icons/$$THEME/$$SIZE/$$CONTEXT/$$ICONFILE; \
+		rm -f $(DESTDIR)$(datadir)/icons/mate/$$SIZE/$$CONTEXT/$$ICONFILE; \
 	done; \
 	for icon in $(private_icons); do \
 		THEME=`echo $$icon | cut -d_ -f1`; \
 		CONTEXT=`echo $$icon | cut -d_ -f2`; \
 		SIZE=`echo $$icon | cut -d_ -f3`; \
 		ICONFILE=`echo $$icon | cut -d_ -f4`; \
-		rm -f $(DESTDIR)$(pkgdatadir)/icons/$$THEME/$$SIZE/$$CONTEXT/$$ICONFILE; \
+		rm -f $(DESTDIR)$(pkgdatadir)/icons/mate/$$SIZE/$$CONTEXT/$$ICONFILE; \
 	done
 
 install-data-local: install-icons update-icon-cache
diff -upr mate-menu-editor-2011.12.01-orig/data/icons/Makefile.in mate-menu-editor-2011.12.01/data/icons/Makefile.in
--- mate-menu-editor-2011.12.01-orig/data/icons/Makefile.in	2011-12-14 11:04:30.000000000 +0100
+++ mate-menu-editor-2011.12.01/data/icons/Makefile.in	2012-03-03 12:27:40.000000000 +0100
@@ -387,12 +387,12 @@ update-icon-cache:
 	@-if test -z "$(DESTDIR)"; then \
 		echo "Updating Gtk icon cache."; \
 		for theme in $(public_icons_themes); do \
-			$(gtk_update_icon_cache) $(datadir)/icons/$$theme; \
+			$(gtk_update_icon_cache) $(datadir)/icons/mate; \
 		done; \
 	else \
 		echo "*** Icon cache not updated.  After (un)install, run this:"; \
 		for theme in $(public_icons_themes); do \
-			echo "***   $(gtk_update_icon_cache) $(datadir)/icons/$$theme"; \
+			echo "***   $(gtk_update_icon_cache) $(datadir)/icons/mate"; \
 		done; \
 	fi
 
@@ -402,16 +402,16 @@ install-icons:
 		CONTEXT=`echo $$icon | cut -d_ -f2`; \
 		SIZE=`echo $$icon | cut -d_ -f3`; \
 		ICONFILE=`echo $$icon | cut -d_ -f4`; \
-		mkdir -p $(DESTDIR)$(datadir)/icons/$$THEME/$$SIZE/$$CONTEXT; \
-		$(INSTALL_DATA) $(srcdir)/$$icon $(DESTDIR)$(datadir)/icons/$$THEME/$$SIZE/$$CONTEXT/$$ICONFILE; \
+		mkdir -p $(DESTDIR)$(datadir)/icons/mate/$$SIZE/$$CONTEXT; \
+		$(INSTALL_DATA) $(srcdir)/$$icon $(DESTDIR)$(datadir)/icons/mate/$$SIZE/$$CONTEXT/$$ICONFILE; \
 	done; \
 	for icon in $(private_icons); do \
 		THEME=`echo $$icon | cut -d_ -f1`; \
 		CONTEXT=`echo $$icon | cut -d_ -f2`; \
 		SIZE=`echo $$icon | cut -d_ -f3`; \
 		ICONFILE=`echo $$icon | cut -d_ -f4`; \
-		mkdir -p $(DESTDIR)$(pkgdatadir)/icons/$$THEME/$$SIZE/$$CONTEXT; \
-		$(INSTALL_DATA) $(srcdir)/$$icon $(DESTDIR)$(pkgdatadir)/icons/$$THEME/$$SIZE/$$CONTEXT/$$ICONFILE; \
+		mkdir -p $(DESTDIR)$(pkgdatadir)/icons/mate/$$SIZE/$$CONTEXT; \
+		$(INSTALL_DATA) $(srcdir)/$$icon $(DESTDIR)$(pkgdatadir)/icons/mate/$$SIZE/$$CONTEXT/$$ICONFILE; \
 	done
 
 uninstall-icons:
@@ -420,14 +420,14 @@ uninstall-icons:
 		CONTEXT=`echo $$icon | cut -d_ -f2`; \
 		SIZE=`echo $$icon | cut -d_ -f3`; \
 		ICONFILE=`echo $$icon | cut -d_ -f4`; \
-		rm -f $(DESTDIR)$(datadir)/icons/$$THEME/$$SIZE/$$CONTEXT/$$ICONFILE; \
+		rm -f $(DESTDIR)$(datadir)/icons/mate/$$SIZE/$$CONTEXT/$$ICONFILE; \
 	done; \
 	for icon in $(private_icons); do \
 		THEME=`echo $$icon | cut -d_ -f1`; \
 		CONTEXT=`echo $$icon | cut -d_ -f2`; \
 		SIZE=`echo $$icon | cut -d_ -f3`; \
 		ICONFILE=`echo $$icon | cut -d_ -f4`; \
-		rm -f $(DESTDIR)$(pkgdatadir)/icons/$$THEME/$$SIZE/$$CONTEXT/$$ICONFILE; \
+		rm -f $(DESTDIR)$(pkgdatadir)/icons/mate/$$SIZE/$$CONTEXT/$$ICONFILE; \
 	done
 
 install-data-local: install-icons update-icon-cache
diff -upr mate-menu-editor-2011.12.01-orig/data/Makefile.am mate-menu-editor-2011.12.01/data/Makefile.am
--- mate-menu-editor-2011.12.01-orig/data/Makefile.am	2011-12-14 11:04:30.000000000 +0100
+++ mate-menu-editor-2011.12.01/data/Makefile.am	2012-03-03 12:41:35.000000000 +0100
@@ -2,11 +2,11 @@ SUBDIRS = icons
 
 @INTLTOOL_DESKTOP_RULE@
 
-desktopdir = $(datadir)/applications
+desktopdir = $(datadir)/applications/mate
 desktop_in_files = alacarte.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 
-pkgdata_DATA = alacarte.ui
+pkgdata_DATA = matealacarte.ui
 
 CLEANFILES = $(desktop_DATA)
 
diff -upr mate-menu-editor-2011.12.01-orig/data/Makefile.in mate-menu-editor-2011.12.01/data/Makefile.in
--- mate-menu-editor-2011.12.01-orig/data/Makefile.in	2011-12-14 11:04:30.000000000 +0100
+++ mate-menu-editor-2011.12.01/data/Makefile.in	2012-03-03 12:41:33.000000000 +0100
@@ -239,10 +239,10 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 SUBDIRS = icons
-desktopdir = $(datadir)/applications
+desktopdir = $(datadir)/applications/mate
 desktop_in_files = alacarte.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
-pkgdata_DATA = alacarte.ui
+pkgdata_DATA = matealacarte.ui
 CLEANFILES = $(desktop_DATA)
 EXTRA_DIST = $(pkgdata_DATA)
 all: all-recursive
diff -upr mate-menu-editor-2011.12.01-orig/Makefile.am mate-menu-editor-2011.12.01/Makefile.am
--- mate-menu-editor-2011.12.01-orig/Makefile.am	2011-12-14 11:04:30.000000000 +0100
+++ mate-menu-editor-2011.12.01/Makefile.am	2012-03-03 12:32:34.000000000 +0100
@@ -4,16 +4,16 @@ SUBDIRS = data po Alacarte
 
 CLEANFILES=
 
-bin_SCRIPTS = alacarte
-CLEANFILES += alacarte
+bin_SCRIPTS = matealacarte
+CLEANFILES += matealacarte
 
 DISTCLEANFILES = ChangeLog
 
-alacarte: alacarte.in
+matealacarte: alacarte.in
 	sed -e s!\@PYTHON\@!@PYTHON@!		\
 	    -e s!\@PYOPTIONS\@!-OOt!		\
 	    < $< > $@
-alacarte: Makefile
+matealacarte: Makefile
 
 EXTRA_DIST = \
 	alacarte.in \
diff -upr mate-menu-editor-2011.12.01-orig/Makefile.in mate-menu-editor-2011.12.01/Makefile.in
--- mate-menu-editor-2011.12.01-orig/Makefile.in	2011-12-14 11:04:30.000000000 +0100
+++ mate-menu-editor-2011.12.01/Makefile.in	2012-03-03 12:32:34.000000000 +0100
@@ -253,8 +253,8 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 SUBDIRS = data po Alacarte
-CLEANFILES = alacarte
-bin_SCRIPTS = alacarte
+CLEANFILES = matealacarte
+bin_SCRIPTS = matealacarte
 DISTCLEANFILES = ChangeLog
 EXTRA_DIST = \
 	alacarte.in \
@@ -785,11 +785,11 @@ uninstall-am: uninstall-binSCRIPTS
 	uninstall uninstall-am uninstall-binSCRIPTS
 
 
-alacarte: alacarte.in
+matealacarte: alacarte.in
 	sed -e s!\@PYTHON\@!@PYTHON@!		\
 	    -e s!\@PYOPTIONS\@!-OOt!		\
 	    < $< > $@
-alacarte: Makefile
+matealacarte: Makefile
 
 ChangeLog:
 	@echo Creating $@
