diff -upr mate-desktop-1.2.0-orig/libmate-desktop/display-name.c mate-desktop-1.2.0/libmate-desktop/display-name.c
--- mate-desktop-1.2.0-orig/libmate-desktop/display-name.c	2012-04-23 19:53:31.000000000 +0200
+++ mate-desktop-1.2.0/libmate-desktop/display-name.c	2012-04-24 12:48:51.000000000 +0200
@@ -209,7 +209,7 @@ read_pnp_ids (void)
         for (i = 0; lines[i]; i++)
         {
              line = lines[i];
-             if (line[3] == '\t')
+             if (line[0] && line[1] && line[2] && line[3] == '\t' && line[4])
              {
                  code = line;
                  line[3] = '\0';
