diff -upr mate-bluetooth-1.2.1-orig/applet/mate-bluetooth-applet.desktop.in.in mate-bluetooth-1.2.1/applet/mate-bluetooth-applet.desktop.in.in
--- mate-bluetooth-1.2.1-orig/applet/mate-bluetooth-applet.desktop.in.in	2012-03-26 01:17:51.000000000 +0200
+++ mate-bluetooth-1.2.1/applet/mate-bluetooth-applet.desktop.in.in	2012-07-01 19:18:34.427849031 +0200
@@ -1,13 +1,9 @@
 [Desktop Entry]
-_Name=Bluetooth Manager
-_Comment=MATE Bluetooth Manager applet
+Name=Bluetooth Manager
+Comment=MATE Bluetooth Manager applet
 Icon=bluetooth
 Exec=mate-bluetooth-applet
 Terminal=false
 Type=Application
 Categories=
-OnlyShowIn=MATE;XFCE;
-X-MATE-Bugzilla-Bugzilla=MATE
-X-MATE-Bugzilla-Product=mate-bluetooth
-X-MATE-Bugzilla-Component=applet
-X-MATE-Bugzilla-Version=@VERSION@
+OnlyShowIn=MATE;
