diff -upr caja-1.2.0-orig/data/caja-folder-handler.desktop.in.in caja-1.2.0/data/caja-folder-handler.desktop.in.in
--- caja-1.2.0-orig/data/caja-folder-handler.desktop.in.in	2012-02-23 13:25:09.000000000 +0100
+++ caja-1.2.0/data/caja-folder-handler.desktop.in.in	2012-05-02 19:09:55.000000000 +0200
@@ -7,7 +7,7 @@ Terminal=false
 Icon=folder-open
 StartupNotify=true
 Type=Application
-MimeType=x-directory/mate-default-handler;x-directory/normal;inode/directory;application/x-mate-saved-search;
+MimeType=inode/directory;application/x-mate-saved-search;
 OnlyShowIn=MATE;
 X-MATE-Bugzilla-Bugzilla=MATE
 X-MATE-Bugzilla-Product=caja
